GoogleChromeLabs/ps-analysis-tool
# Fix: Merge blocked reasons data in dashboard

## Description
This PR adds code to merge blocked reasons and `isBlocked` key data across the occurrence in page URLs for similar cookies.
Alongside updates the condition for filtering cookies with issues.


## Relevant Technical Choices
- The function `extractReportData` parses the CLI output JSON file and merges cookies into a single object, where repeating cookies are reassigned to the key-value pair.
- In the sitemap landing page showing the count of cookies and their categories, we need unified(non-repeating cookies) data from all the pages.
- This was working until blockedReasons were required to be shown on the landing page, so due to reassigning the blockedReasons were not preserved across the repetition of cookies for sitemap data.
- During that implementation few places were missed for modification in data processing logic.
- To tackle this, while calculating cookies for landing page count previous data is preserved, alongside using the blockedReasons key for showing blocked cookies in the table.


## Testing Instructions
- Build CLI on `develop` and run sitemap https://rtcamp.com/sitemap.xml.
- Choose 2 sites for analysis, once done open the dashboard.
- You'll notice that the sitemap landing page has no blocked cookies, but as you go through each site individually, blocked cookies should be there.
- Now switch to this branch and repeat the same steps.
- You'll notice counts are correctly being shown on the sitemap landing page.


## Additional Information:



## Screenshot/Screencast



---

## Checklist






Fixes #
